mmengine/dataset/utils.py

```diff
@@ -16,11 +16,11 @@
 COLLATE_FUNCTIONS = FUNCTIONS
 
 
-def worker_init_fn(worker_id: int,
-                   num_workers: int,
-                   rank: int,
-                   seed: int,
-                   disable_subprocess_warning: bool = False) -> None:
+def default_worker_init_fn(worker_id: int,
+                           num_workers: int,
+                           rank: int,
+                           seed: int,
+                           disable_subprocess_warning: bool = False) -> None:
```

Comment:
Suggest keeping the name of worker_init_fn to avoid BC-breaking

Reply:
Ok, I will fix it and I found the main branch can't pass the pre-commit, so how can I deal with it?

Reply:
Do you mean the lint error like this:

![image](https://user-images.githubusercontent.com/57566630/229743343-6c06a29f-43a1-423d-bc79-bbff9a74ea0c.png)

Since the `worker_init_fn` is got by `Registry.get` now, it could be a None object, which is not acceptable for partial. You could use `assert `init_wroker_fn` is not None` to narrow its type and pass the mypy check.

Reply:
There are many errors in other modules.

![2023-04-04 19-36-28屏幕截图](https://user-images.githubusercontent.com/26410476/229780136-5a0ce813-6151-406d-bd9f-f12a91c813c5.png)